Build failure with gcc 12

We have the following build failures on buildroot autobuilders with gcc 12:

```
/tmp/instance-8/output-1/build/hawktracer-2ec19d71923344148928ef45fce549eda67030a4/hawktracer.cpp: In function 'HT_Thread* ht_thread_create(ht_thread_callback_t, void*)':
/tmp/instance-8/output-1/build/hawktracer-2ec19d71923344148928ef45fce549eda67030a4/hawktracer.cpp:1006:28: error: no matching function for call to 'operator new(sizetype, std::thread*)'
 1006 |     new(&th->th) HT_Thread();
      |                            ^
```

Full build log: http://autobuild.buildroot.org/results/d18/d18a1b2ee6cdcb289567ae90b2d836844427b5f6/build-end.log

However, I don't know how to fix it properly, any advise is appreciated.

--- a/lib/tcp_server.cpp
+++ b/lib/tcp_server.cpp
@@ -14,6 +14,7 @@
 
 #if !defined(HT_THREAD_FORCE_SELECTED) && defined(HT_CPP11)
 #  include <thread>
+#  include <new>
 #  define HT_THREAD_IMPL_CPP11
 #elif !defined(HT_THREAD_FORCE_SELECTED) && defined(_WIN32)
 #  include <windows.h>

